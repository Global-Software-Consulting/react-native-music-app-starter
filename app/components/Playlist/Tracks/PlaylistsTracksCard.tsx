import React from 'react';
import { View, TouchableOpacity, Image } from 'react-native';
import { Text } from 'react-native-paper';
import useStyles from '../PlaylistSongs/styles';
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';

interface MusicProps {
    name?: string;
    model?: string;
    img?: string;
    item?: any;
    setSong?: any;
    playlistRef?: any;
    onPress?: any;
}

const PlaylistsTracksCard: React.FC<MusicProps> = ({
    name,
    model,
    img,
    setSong,
    item,
    playlistRef,
    onPress,
}) => {
    const styles = useStyles();
    return (
        <TouchableOpacity onPress={onPress} style={{ flexDirection: 'row' }}>
            <View style={styles.container}>
                <TouchableOpacity style={styles.taskCard}>
                    <View style={styles.imgcontainer}>
                        <Image style={styles.img} source={{ uri: img }} />
                    </View>
                </TouchableOpacity>
                <View style={styles.nameContainer}>
                    <View style={styles.textWrapper}>
                        <Text style={styles.label}>{name}</Text>
                        <Text style={styles.model}>{model} </Text>
                    </View>
                </View>
                <View style={{ width: '15%' }} />
                <TouchableOpacity
                    onPress={() => {
                        setSong(item);
                        playlistRef.current.snapToIndex(0);
                    }}>
                    <MaterialCommunityIcons name="dots-vertical" size={25} color={'gray'} />
                </TouchableOpacity>
            </View>
        </TouchableOpacity>
    );
};

export default PlaylistsTracksCard;
