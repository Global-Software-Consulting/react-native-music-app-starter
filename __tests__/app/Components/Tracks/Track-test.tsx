/**
 * @format
 */
import 'react-native';
import React from 'react';
import PlaylistsTracksCard from '../../../../app/components/Playlist/Tracks/PlaylistsTracksCard';
// Note: test renderer must be required after react-native.
import { render, fireEvent } from '@testing-library/react-native';
import { CurrentRenderContext } from '@react-navigation/native';


jest.mock("react-native-responsive-screen", () => {
    return {
        heightPercentageToDP: jest.fn(),
        widthPercentageToDP: jest.fn(),

    }
});

jest.mock('@react-navigation/native', () => {
    return {
        useIsFocused: jest.fn(),
        useNavigation: jest.fn(),
        useRoute: jest.fn(),

    }


});
jest.mock('react-i18next', () => {
    return {
        useTranslation: () => {
            return {
                t: jest.fn(),
            };
        },
    };
});

jest.mock('react-native-iphone-x-helper', () => ({
    getStatusBarHeight: jest.fn(),
    getBottomSpace: jest.fn(),
}));
jest.mock('react-redux', () => ({
    useDispatch: jest.fn(),
    useSelector: jest.fn(),
}));

describe("group tesing goes here", () => {
    test('adding a songs', () => {
        // const playlistReference = { playlistRef: { current: { snapToIndex: 0 } } };
        const { getByTestId } = render(<PlaylistsTracksCard />);
        fireEvent.press(getByTestId('setASongs'));
    });
    test('setting songs', () => {
        const playlistReference = { current: { snapToIndex: 0 } };
        const { getByTestId } = render(<PlaylistsTracksCard setSong={() => { }} playlistRef={playlistReference} />);
        fireEvent.press(getByTestId('settingSongs'));
    });
    // test('card task', () => {
    //     const { getByTestId } = render(<PlaylistsTracksCard />);
    //     fireEvent.press(getByTestId('cardTask'));
    // });

})

