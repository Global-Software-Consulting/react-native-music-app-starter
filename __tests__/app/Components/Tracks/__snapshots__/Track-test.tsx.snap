// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`group tesing goes here renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#f6f6f6",
      "flexDirection": "row",
      "justifyContent": "center",
      "marginBottom": 10,
    }
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#f6f6f6",
        "borderRadius": 12,
        "flexDirection": "row",
        "justifyContent": "center",
        "opacity": 1,
        "paddingLeft": 5,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#f6f6f6",
          "borderRadius": 10,
          "elevation": 20,
          "flexDirection": "row",
          "height": undefined,
          "shadowColor": "#f6f6f6",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.2,
          "shadowRadius": 8,
          "width": undefined,
        }
      }
    >
      <Image
        source={Object {}}
        style={
          Object {
            "backgroundColor": "#f6f6f6",
            "borderRadius": 5,
            "height": undefined,
            "resizeMode": "cover",
            "width": undefined,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "flexDirection": "row",
        "height": undefined,
        "justifyContent": "center",
        "width": undefined,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "height": undefined,
          "justifyContent": "flex-start",
          "width": undefined,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "textAlign": "left",
            },
            Object {
              "backgroundColor": "#f6f6f6",
              "fontSize": undefined,
              "fontWeight": "bold",
              "height": undefined,
              "marginBottom": 12,
              "width": undefined,
            },
          ]
        }
      />
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "fontFamily": "System",
              "fontWeight": "400",
            },
            Object {
              "textAlign": "left",
            },
            Object {
              "backgroundColor": "#f6f6f6",
              "borderRadius": 10,
              "color": "darkgray",
              "fontSize": undefined,
              "height": undefined,
              "justifyContent": "center",
              "marginTop": -7,
              "textAlign": "center",
              "width": undefined,
            },
          ]
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "width": "15%",
      }
    }
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <mockedDefaultExport
      color="gray"
      name="dots-vertical"
      size={25}
    />
  </View>
</View>
`;
